src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -202,11 +205,37 @@ public function supportsDenormalization($data, $type, $format = null, array $con
      */
     private function getNormalizer($data, $format, array $context)
     {
+        $type = \is_object($data) ? 'c-'.\get_class($data) : \gettype($data);
```

Comment:
does it make any sense to cache by internal type?
I feel like only classes should allow caching but I might be wrong
if the answer is yes, I'd suggest using the prefix on the gettype side, would be better for performance.

Reply:
what it's doing is caching the fact that no normalizers supported a scalar, which allows a shortcut to escape testing again.